Formula/makensis.rb

```diff
@@ -43,6 +44,7 @@ def install
 
   test do
     system "#{bin}/makensis", "-VERSION"
-    system "#{bin}/makensis", "#{share}/nsis/Examples/bigtest.nsi", "-XOutfile /dev/null"
+    # Force ASCII target to avoid Unicode issue (until resolved upstream), see https://sourceforge.net/p/nsis/bugs/1165/
+    system "#{bin}/makensis", "-XUnicode false", "#{share}/nsis/Examples/bigtest.nsi", "-XOutfile /dev/null"
```

Comment:
Does this mean any use of `makensis` with the new version requires the flag `-XUnicode false` at the moment? I'm a little wary of shipping something to users that requires a workaround they may or may not be aware of.

When this is working properly, when does a user want to set `-XUnicode {true,false}`?

Reply:
Yes, the situation is terrible and the way I see it, the alternatives are even worse:

1. Skip this version, until the bug was fixed. This ignores the fact, that the bug already exists in previous versions that can be installed with Homebrew (see table below).
2. Build `makensis` without Unicode support enabled. This is hard to communicate to users and might not meet their expectations.
3. Patch the source code. Since upstream is yet unaware on how to fix this, it's unrealistic that we can do it.

To be sure that everybody understands the change that's comes with this version:

| Unicode Setting | NSIS version | Compiles |
|-----------------|--------------|----------|
| (not set)       | <3.07        | ✅       |
| `false`         | <3.07        | ✅       |
| `true`          | <3.07        | ❌        |
| (not set)       | >=3.07       | ❌        |
| `false`         | >=3.07       | ✅       |
| `true`          | >=3.07       | ❌        |
    
The situation is worse for users that implicitly set Unicode to `true`. For users that explicitely set Unicode to either `true` or `false`, the situation does not change compared to older NSIS versions.

Reply:
Earlier versions are still a little different in an important way: if you don't use `-XUnicode`, then this should work. Now, it won't work _unless_ you use `-XUnicode false`.

I'm still interested in the answer to my question above:
> When this is working properly, when does a user want to set `-XUnicode {true,false}`?

Reply:
> Earlier versions are still a little different in an important way: if you don't use -XUnicode, then this should work. Now, it won't work unless you use -XUnicode false

Exactly, that's because implicit usage of the `Unicode` keyword has changed (it now defaults to `true`), but explicit usage remains the same. In either case, `Unicode true` will fail. Users _still_ cannot compile Unicode NSIS installers, as it's been the case with earlier versions of NSIS, at least under certain conditions (see the SourceForge issue I linked earlier).

Reply:
It depends. What exactly does the `-XUnicode` flag do, and when does a user want to set it to `true` or `false`?

Reply:
> Starting with NSIS v3.0 you can choose to create Unicode installers by setting the Unicode attribute. These installers will not work on Windows 95/98/ME but they will allow you to display your installer in any Unicode language supported by the OS.
> 
> When building a Unicode installer NSIS variables can hold Unicode characters (0001-FFFF). There should be no need to modify your existing scripts. If you want to read/write Unicode files, specific instructions have been added to read/write UTF-16LE strings from/to disk.
>
> [via](https://nsis.sourceforge.io/Docs/Chapter1.html#intro-unicode)

Reply:
Hmm... Let's find out.

Reply:
I think we found our culprit:
```
==> locale
LANG=""
LC_COLLATE="C"
LC_CTYPE="C"
LC_MESSAGES="C"
LC_MONETARY="C"
LC_NUMERIC="C"
LC_TIME="C"
LC_ALL=
==> id
uid=502(brew) gid=20(staff) groups=20(staff),12(everyone),61(localaccounts),702(com.apple.sharepoint.group.2),100(_lpoperator),703(com.apple.sharepoint.group.3),701(com.apple.sharepoint.group.1)
```

Reply:
I'm happy to hear this. Thanks for you support!

Reply:
No problem. You did all the hard work. Thanks for your PR!